src/views/home/components/bridge-form/bridge-form.view.tsx

```diff
@@ -284,29 +297,30 @@ const BridgeForm: FC<BridgeFormProps> = ({ account, formData, resetForm, onSubmi
     /*
      * Load the default values after the network is changed
      */
-    if (env && connectedProvider && formData === undefined) {
+    if (env && connectedProvider) {
```

Comment:
We have a big race condition between this `useEffect` hook and the one below. In fact, this is working because how hooks are placed. If you try to move this hook below the one below, you'll see that it will not work as it will set the amount to `undefined` just after correctly set the `amount` to the desired value from the `formData`.

We should investigate how can we improve this.

Reply:
Yep. We already discussed this in the past and found no simple solution. Do you have new ideas?